jax/core.py

```diff
@@ -1203,6 +1203,9 @@ def type_transfer(prim, invars, params):
       if "mapped_invars" not in params:
         raise TypeError(
             f"Map primitive {prim} missing 'mapped_invars' parameter")
+      if "donated_invars" not in params:
```

Comment:
I think this check should be hoisted one level up, i.e. it shouldn't only apply to the `if prim.map_primitive` case but rather to the `if prim.call_primitive or prim.map_primitive` case.

Reply:
Happy to do so, but we'd need to match on primitive name which doesn't feel ideal, since donation doesn't apply to all `call` primitives (e.g. remat does not have this argument) but it does apply to all (well, the only) `map` primitive.

Reply:
We might want `if prim.consuming_primitive` or something like that, but no need to do that now.

Reply:
Oh yeah, good point, I forgot! Nevermind.